src/test/ui/issues/issue-18446.stderr

```diff
@@ -2,7 +2,10 @@ error[E0034]: multiple applicable items in scope
   --> $DIR/issue-18446.rs:18:7
    |
 LL |     x.foo();
-   |       ^^^ multiple `foo` found
+   |     --^^^--
+   |     | |
+   |     | multiple `foo` found
+   |     help: disambiguate the method call for candidate #2: `T::foo(&x)`
```

Comment:
Why is there only one disambiguation suggestion here?

Reply:
The code for this is

```

trait T {
    fn foo(&self);
}

impl<'a> dyn T + 'a {
    fn foo(&self) {}
}

impl T for i32 {
    fn foo(&self) {}
}

fn main() {
    let x: &dyn T = &0i32;
    x.foo(); //~ ERROR multiple applicable items in scope [E0034]
}
```

I'm not sure what the appropriate syntax would be to call `&dyn T::foo`. Also note that we were _already_ not suggesting it.

Reply:
> Also note that we were already not suggesting it.

I did notice this. I think we ought to open a new issue — if the user can't specify `(&dyn T)::foo`, we should at least explain this in the diagnostic. But this PR looks good as it is.